cli/pcluster/cli_commands/delete.py

```diff
@@ -117,5 +120,41 @@ def _delete_cluster(cluster_name, nowait):
         sys.stdout.flush()
         sys.exit(1)
     except KeyboardInterrupt:
+        terminate_compute_fleet = False
         LOGGER.info("\nExiting...")
         sys.exit(0)
+    finally:
+        if terminate_compute_fleet:
+            _terminate_cluster_nodes(stack_name)
+
+
+def _terminate_cluster_nodes(stack_name):
+    try:
+        LOGGER.debug("Compute fleet clean-up: STARTED")
+        LOGGER.info("\nChecking for running EC2 hosts that require termination")
+        ec2 = boto3.client("ec2", config=Config(retries={"max_attempts": 10}))
+
+        for instance_ids in _describe_instance_ids_iterator(stack_name):
+            LOGGER.debug("Terminating instances %s", instance_ids)
+            if instance_ids:
+                ec2.terminate_instances(InstanceIds=instance_ids)
+
+        LOGGER.debug("Compute fleet clean-up: COMPLETED")
+    except Exception as e:
+        LOGGER.error("Failed when terminating EC2 instances with error %s", e)
+
+
+def _describe_instance_ids_iterator(stack_name, instance_state=("pending", "running", "stopping", "stopped")):
+    ec2 = boto3.client("ec2")
+    filters = [
+        {"Name": "tag:Application", "Values": [stack_name]},
+        {"Name": "instance-state-name", "Values": list(instance_state)},
+        {"Name": "tag:aws-parallelcluster-node-type", "Values": [str(NodeType.compute)]},
+    ]
+    paginator = ec2.get_paginator("describe_instances")
```

Comment:
Could have we used the utility `paginate_boto3` here?

Reply:
yes I can switch to that

Reply:
done